### DISCUSSION TITLE
Incorrect cursor movement using OpenTabletDriver with osu versions > `2022.509.0`

### DISCUSSION THREAD

**Discussion by NoraPerry** [ppy/osu#19619] [Q&A]
### Type

Game behaviour

### Bug description

When using OpenTabletDriver with osu versions newer than `2022.509.0`, osu moves the cursor as if the tablet is in relative mode instead of absolute.

The cursor moves correctly outside of osu with otd, and the built in driver in osu also works correctly.

I am using the osu AppImage on arch linux with sway wm.

### Screenshots or videos

I've attached screenshots of the tablet settings on otd and osu. I figured a video wouldn't be useful because it'd just be a video of the cursor moving around but I can capture one if it would be helpful.

![osu](https://user-images.githubusercontent.com/43629496/183274507-09e87dee-5fed-41e9-887b-88760a85b7b6.png)
![otd](https://user-images.githubusercontent.com/43629496/183274509-bea59971-1e75-4525-8c83-9494db44e6fe.png)

### Version

Above 2022.509.0

### Logs

Logs are from the latest osu version at the time of writing. I just opened osu, moved the cursor around a bit then closed it.

[database.log](https://github.com/ppy/osu/files/9276151/database.log)
[network.log](https://github.com/ppy/osu/files/9276152/network.log)
[performance.log](https://github.com/ppy/osu/files/9276153/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/9276154/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/9276155/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/9276156/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/9276157/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/9276158/runtime.log)

**Comment by peppy** [comment]
Can you try turning off touch support and possible high precision mouse support in-game? It sounds like something must be reporting tablet input as relative touch input or similar.

**Reply by NoraPerry** [reply]
It behaves the same way whether touch and high precision mouse are toggled on or off.

**Comment by NoraPerry** [comment]
I git bisected and the first bad commit is 3cbd19a9ce09c800005e6b51d263e4d4593362d5 which is just updating the framework version. I'll see if I can bisect the framework to find the issue.

**Reply by frenzibyte** [reply]
Will probably be either https://github.com/ppy/osu-framework/pull/5136 or https://github.com/ppy/osu-framework/pull/5163.

**Reply by NoraPerry** [reply]
It was https://github.com/ppy/osu-framework/pull/5136, thanks for the hint. The first bad commit in the framework is [841e2ccc192dea70978a18c7dd52ffa3d8ec38e5](https://github.com/ppy/osu-framework/commit/841e2ccc192dea70978a18c7dd52ffa3d8ec38e5) which is updating the SDL package.

I'm not really sure how to go about debugging the SDL repo but I'm guessing the problem was introduced in the commits on May 9th to https://github.com/ppy/SDL2-CS. I doubt adding the const strings or adding the gfx wrapper would have broken anything so is it possible it's a problem with SDL itself?

**Reply by frenzibyte** [reply]
This should represent a list of all commits that have been pushed since the SDL build that was used prior to ppy/osu-framework#5136, taken by the dates of the SDL2-CS versions: https://github.com/libsdl-org/SDL/compare/3e43d6e34b356f7646c988459748d1f7b56d6864...85977354fbce9ecb2add660a88e6bec3532c3e26

Next step would be to isolate this issue and start bisecting on the above commit range, or at the very least report the issue to SDL.

**Reply by peppy** [reply]
Also of note, SDL has been updated on the framework side to the latest version, which may come with a fix for this. A good first step before expending too much energy would be to `UseLocalFramework` with the latest framework `master` branch and see if the issue is fixed.

**Reply by NoraPerry** [reply]
> Also of note, SDL has been updated on the framework side to the latest version, which may come with a fix for this. A good first step before expending too much energy would be to `UseLocalFramework` with the latest framework `master` branch and see if the issue is fixed.

Just tried that with osu and osu-framework both on latest master branch and it's not fixed.
